### ISSUE TITLE
After a successful warp sync, parity does not download ancient blocks

### ISSUE THREAD

**Issue by vrde** [paritytech/parity-ethereum#9853]
- **Parity Ethereum version**: v.2.0.9 ~~, v2.1.4-beta~~
- **Operating system**: GNU/Linux
- **Installation**: one-line installer
- **Fully synchronized**: yes
- **Network**: ethereum
- **Restarted**: ?

Disclaimer: I'm using an HDD, not an SSD, but the disk IO seems to be reasonably under control.

I'm able to warp sync and reach a fully sync state in my node. I guess I [understand the basics on how the warp sync works](https://gitter.im/paritytech/parity?at=5bdb48c22cb7054a8cf4de36).

Since my node is 100% in sync, I expect it to start downloading the *ancient blocks*, but I guess this is not happening. The logs don't mention any download of those blocks. If I manually query the JSONRPC interface I see all blocks between `0` and `57`, and from `6615001` to `latest`. I can create a script to check if there are other blocks in the interval `58, 6615000`, but I didn't, so I cannot say I don't have **any** blocks in this range.

I've moved out from v2.1.3-beta and sync from scratch, as [suggested](https://twitter.com/5chdn/status/1057953879214735363).

I've tried to remove `nodes.json` as [suggested](https://gitter.im/paritytech/parity?at=5bdbfb87cd2a6b5bdc82101c).

I've also tried to use the `parity_chainStatus` method, but in both cases (v.2.0.9, v2.1.4-beta) the `blockGap` is `null`.
```
ubuntu@ryan9000:/data/parity$ curl --data '{"method":"parity_chainStatus","params":[],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" -X POST localhost:8545

{"jsonrpc":"2.0","result":{"blockGap":null},"id":1}
```

Note 0: I synced from scratch using v2.0.9, and now I'm running v2.1.4-beta using the database created by v2.0.9.

Note 1: I'm aware it's "difficult" to sync with an HDD, but I have time.

Note 2: If you want to inspect my machine I can add your GitHub pubkey to my ssh server.

Note 3: I'm aware of the great blog post by Afri [The Ethereum-blockchain size will not exceed 1TB anytime soon. - DEV Community 👩‍💻👨‍💻](https://dev.to/5chdn/the-ethereum-blockchain-size-will-not-exceed-1tb-anytime-soon-58a).

Note 4: You can check the status of my node from here http://ryan9000.ddns.net:8000/ (it's not a VM in the cloud, so it might go offline sometimes).

**Comment by ascjones** [comment]
> Note 0: I synced from scratch using v2.0.9, and now I'm running v2.1.4-beta using the database created by v2.0.9.

v2.1.4-beta contains the fix for this bug, but will only work if you sync from scratch with this version.

**Comment by vrde** [comment]
@ascjones I'm genuinely trying to understand as much as possible so I can avoid repeating the same mistakes in the future, and please don't read this comment as non-constructive.

To give context: after knowing that *v2.1.3-beta* was broken, I decided to use the stable version of parity to avoid other problems so I reset everything and sync with *2.0.9-stable*.

Both *2.0.9-stable* and *2.1.4-beta* have been released 4 days ago, so what's the purpose of *2.0.9-stable*? Is it something I can use if and only if my node was synchronized with a previous, stable version of parity?

One last question: if I sync with *2.1.4-beta* and I want to upgrade to a newer (beta|stable) version, do I have to sync from scratch again?

**Comment by joshua-mir** [comment]
It's my understanding that you can upgrade the database, but downgrading isn't supported. You can't upgrade from any 2.1 track release to a 2.0 track release, even if it is more recent, without resyncing, but the 2.1 release track will eventually be marked stable in a later version, and you will be upgrading and won't need to resync, even though you'll be moving from beta to stable.

**Comment by ascjones** [comment]
Also, the fix for this specific issue (not downloading ancient blocks) is included in 2.1.4-beta but *not* in 2.0.9-stable. And the fix will only work when syncing from a fresh database, which is why I made the suggestion. So this is separate from general DB compatibility between release tracks/versions.

Hope that's not too confusing!

**Comment by vrde** [comment]
About *2.0.9-stable*, my only point was that if you start a sync from scratch ancient blocks won't be downloaded, so I was wondering why releasing a new stable version if it has a (known?) bug.

Thanks for the support, by using *2.1.4-beta* I was able to warp sync **and** to download ancient blocks. I think I read somewhere that you **must** have an SSD to sync, but I was able to do a full sync in 3 days using an old school HDD.
